### ISSUE TITLE
Band stats on geotiff writer incorrect

### ISSUE THREAD

**Issue by vpipkt** [locationtech/rasterframes#321]
The `geotiff` writer puts four statistics on each band: max, min, mean, stddev. QGIS for example uses the min and max as rendering hints. It will also rely on these for functionality like computing histograms in some way.  

A sample of `gdalinfo` against a 3 band product looks like:

```
Band 1 Block=256x256 Type=UInt16, ColorInterp=Red
  Min=0.000 Max=0.000
  Minimum=0.000, Maximum=0.000, Mean=0.000, StdDev=0.000
  Metadata:
    RF_COL=B7
    STATISTICS_MAXIMUM=0
    STATISTICS_MEAN=0
    STATISTICS_MINIMUM=0
    STATISTICS_STDDEV=0
Band 2 Block=256x256 Type=UInt16, ColorInterp=Green
  Min=0.000 Max=0.000
  Minimum=0.000, Maximum=0.000, Mean=0.000, StdDev=0.000
  Metadata:
    RF_COL=B5
    STATISTICS_MAXIMUM=0
    STATISTICS_MEAN=0
    STATISTICS_MINIMUM=0
    STATISTICS_STDDEV=0
Band 3 Block=256x256 Type=UInt16, ColorInterp=Blue
  Min=0.000 Max=0.000
  Minimum=0.000, Maximum=0.000, Mean=0.000, StdDev=0.000
  Metadata:
    RF_COL=B1
    STATISTICS_MAXIMUM=0
    STATISTICS_MEAN=0
    STATISTICS_MINIMUM=0
    STATISTICS_STDDEV=0
```

Running again as `gdalinfo --mm` which says, compute your own min and max. What GDAL is computing shows up as "Computed Min/Max=". Note the mismatch here. 

```
Band 1 Block=256x256 Type=UInt16, ColorInterp=Red
  Min=0.000 Max=0.000   Computed Min/Max=5036.000,24873.000
  Minimum=0.000, Maximum=0.000, Mean=0.000, StdDev=0.000
  Metadata:
    RF_COL=B7
    STATISTICS_MAXIMUM=0
    STATISTICS_MEAN=0
    STATISTICS_MINIMUM=0
    STATISTICS_STDDEV=0
Band 2 Block=256x256 Type=UInt16, ColorInterp=Green
  Min=0.000 Max=0.000   Computed Min/Max=5769.000,32048.000
  Minimum=0.000, Maximum=0.000, Mean=0.000, StdDev=0.000
  Metadata:
    RF_COL=B5
    STATISTICS_MAXIMUM=0
    STATISTICS_MEAN=0
    STATISTICS_MINIMUM=0
    STATISTICS_STDDEV=0
Band 3 Block=256x256 Type=UInt16, ColorInterp=Blue
  Min=0.000 Max=0.000   Computed Min/Max=9523.000,19506.000
  Minimum=0.000, Maximum=0.000, Mean=0.000, StdDev=0.000
  Metadata:
    RF_COL=B1
    STATISTICS_MAXIMUM=0
    STATISTICS_MEAN=0
    STATISTICS_MINIMUM=0
    STATISTICS_STDDEV=0
```

The bottom line here is that rendering in QGIS seems broken but the real problem is our incorrect tags.

**Comment by vpipkt** [comment]
Ah yes i think that might be true but I am unsure where that is coming
from. Will provide a more thorough replication.

**Comment by vpipkt** [comment]
Closing this issue.... See [attached Jupyter notebook](https://github.com/locationtech/rasterframes/files/3556922/RF-321.ipynb.zip) . The relevant part is the GDAL info on writing to a fresh empty directory:

```
Band 1 Block=256x256 Type=Float64, ColorInterp=Gray
  NoData Value=nan
  Metadata:
    RF_COL=eigth
```

We also confirm no tif.xml is written. Must have been i tried to open a bum file with the same name.

Going to file an issue  to remove a `${output_filename}.xml` sidecar in the destination if it exists
